engine/include/cubos/engine/imgui/data_inspector.hpp

```diff
@@ -0,0 +1,42 @@
+/// @file
+/// @brief Resource @ref cubos::engine::DataInspector.
+/// @ingroup imgui-plugin
+
+#pragma once
+
+#include <cubos/core/reflection/type.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Resource which allows user to inspect or modify object's properties in the UI.
+    ///
+    /// @ingroup imgui-plugin
+    class DataInspector final
+    {
+        /// @brief Alias for @ref core::reflection::Type.
+        using Type = core::reflection::Type;
```

Comment:
I think its best to omit the alias here, its just some extra characters in the methods and it won't ever be used through the class elsewhere.